Scripts/Generator/RoomGenerator.cs

```diff
@@ -128,4 +169,74 @@ public PhysicsBody GenerateWall(Vector3 wallBase, Vector3 size, Vector3 normal)
         wall.Rotation = new Vector3(0, 1, 0) * normal.AngleTo(Vector3.Forward);
         return wall;
     }
+
+    public void GenerateFloor()
+    {
+        if (floor == null)
+            return;
+        Debug.Assert(tileOffset >= 0 && tileOffset <= 1);
+        Debug.Assert(tileSize.Length() > 0);
+        foreach (Node tile in floor.GetChildren())
+        {
+            if (tile != null)
+                tile.QueueFree();
+        }
+        Vector2 roomSpaceSize = roomSize - 2 * wallThickness * Vector2.One;
+        Vector3 startingCorner = origin - roomSpaceSize.x / 2 * left - roomSpaceSize.y / 2 * front;
+        for (int x = 0; x < roomSpaceSize.x / tileSize.x; x++)
+        {
+            float rowTileOffset = (tileOffset * x) % 1;
```

Comment:
`% 1`?

Reply:
So that the tile offset is between 0 and 1 (percentage of tile to offset by), so that it wouldn't make a "staircase" effect.